app/docs/dev.md

```diff
@@ -1,29 +1,207 @@
+
+Note that in lists of groups the items should ideally be written in the form $q_1^{c_1} \cdot q_2^{c_2} \cdots q_n^{c_n}$ where $q_1, q_2 \ldots q_n$ are quantities and $c_1, c_2 \ldots c_n$ are integers, but the function can also handle item that are sums with terms written in the form $a \cdot q_1^{c_1} \cdot q_2^{c_2} \cdots q_n^{c_n}$ where $q_1, q_2 \ldots q_n$ are quantities, $c_1, c_2 \ldots c_n$ rational numbers and $a$ a constant. If the total number of groups is less than required the set of groups is considered invalid, even if there is a sufficient number of terms with independent power products in the response.
+
+The core part of the function is an algorithm that takes two sets of groups and detmines if they generate the same nullspace. If one set of groups is valid and both groups generate the same nullspace then the second set of groups is also valid. It is assumed that the task author either supplies a valid set (or details on the dimensions of the quantities involved such that a valid set can be generated) that can used to check the validity of response. For a more detailed discussion of the algorithm see: [Automated feedback on student attempts to produce a set of dimensionless power products from a list of quantities that describe a physical problem](https://doi.org/10.6084/m9.figshare.24850131), the evaluation implements the algorithm described there (see `determine_validity` in `evaluation.py`) together with some modifications (see below).
+
+To extend the functionality of the evaluation function the algorithm described in the preceding paragraph has been modified in two ways:
+1. If no example set is supplied, but a list of quantities is, then an example of a valid set is constructed by constructing the dimensional matrix corresponding to the list of quantities and picking a set of independent groups that span its nullspace. This example set of groups is then used a reference answer.
+2. If the response contains an expression in the form $a \cdot q_1^{c_1} \cdot q_2^{c_2} \cdots q_n^{c_n}$ where $q_1, q_2 \ldots q_n$ are quantities, $c_1, c_2 \ldots c_n$ rational numbers and $a$ a constant, with more than one term then the evaluation function extracts all the terms treats them as separate elements of a set of groups. For the response to be considered valid it is required that the original number of groups in the response (i.e. before term extraction) is the correct number of groups as given by the Buckingham Pi Theorem.
+The code for these modifications can be found in `evaluation_function` in `evaluation.py`.
+
 
 ## Inputs
-*Specific input parameters which can be supplied when the `eval` command is supplied to this function.*
+All input parameters need to be supplied via the `params` input value to `evaluation_function` in `evaluation.py`.
+
+There are four optional parameters that can be set: `custom_feedback`, `elementary_functions`, `quantities`, `strict_syntax`.
+
+## `custom_feedback`
```

Comment:
As this is the least likely parameter to be used, would it be better at the end of the list of parameters?

Reply:
Parameters are listed in alphabetical order for consistency. Decided to keep this order.